pkg/operator/controller/controller_dns.go

```diff
@@ -14,7 +14,14 @@ import (
 // ensureDNS will create DNS records for the given LB service. If service is
 // nil, nothing is done.
 func (r *reconciler) ensureDNS(ci *operatorv1.IngressController, service *corev1.Service, dnsConfig *configv1.DNS) error {
-	dnsRecords, err := desiredDNSRecords(ci, service, dnsConfig)
+	// If no load balancer has been provisioned, we can't do anything with the
+	// configured DNS zones.
+	ingress := service.Status.LoadBalancer.Ingress
+	if len(ingress) == 0 || len(ingress[0].Hostname) == 0 {
```

Comment:
Can an ingress IP sneak by causing `desiredDNSRecords` to create an empty record? I know we're only using `Hostname` for now, but I wonder if this can lead to a bug in the future. Does it make sense for the condition to only be `len(ingress[0].Hostname) == 0`?